rio/src/RIO/Deque.hs

```diff
@@ -0,0 +1,258 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE TypeFamilies #-}
```

Comment:
I might be missing it, but it doesn't look like `TypeFamilies` is necessary here anymore, same with `DequeSpec.hs` file